src/lib.rs

```diff
@@ -247,18 +243,54 @@ where
 #[derive(Debug)]
 enum Authentication<'a> {
     Password { username: &'a str, password: &'a str },
+    Gssapi { gssapi_authenticator: GssapiAuthenticator<'a>},
     None,
 }
 
+pub struct GssapiAuthenticator<'a> {
+    gssapi_authenticator: &'a dyn GssapiAuthentication,
+    renegotiate_sec_token: bool
+}
+
+impl<'a> GssapiAuthenticator<'a> {
+    pub fn new(gssapi_authenticator: &'a dyn GssapiAuthentication, renegotiate_sec_token: bool) -> Self {
+        GssapiAuthenticator{gssapi_authenticator, renegotiate_sec_token }
+    }
+}
+
+impl <'a>Debug for GssapiAuthenticator<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "GssapiAuthenticator")
+    }
+}
+
 impl<'a> Authentication<'a> {
     fn id(&self) -> u8 {
         match self {
             Authentication::Password { .. } => 0x02,
+            Authentication::Gssapi { .. } => 0x01,
             Authentication::None => 0x00,
         }
     }
 }
 
+#[async_trait]
+pub trait GssapiAuthentication: Send + Sync {
+    // This method retrieves the security context token, 
+    // server_challenge as None:    means return the first init_sec token
+    // server_challenge as Some(x): means return the resp for the servers challenge
+    async fn get_security_context(&self, server_challenge: Option<&[u8]>) -> std::result::Result<String, Error>;
```

Comment:
Maybe `Vec<u8>` as the return value is better because it doesn't have to be UTF-8.

Reply:
yep, seems a good choice. Will do